src/quote/flashmint/nav/provider.ts

```diff
@@ -0,0 +1,117 @@
+import { BigNumber } from '@ethersproject/bignumber'
+
+import { AddressZero } from 'constants/addresses'
+import { USDC } from 'constants/tokens'
+import { SwapQuoteProvider } from 'quote/swap'
+import {
+  Exchange,
+  getFlashMintNavContract,
+  isSameAddress,
+  slippageAdjustedTokenAmount,
+  SwapDataV3,
+} from 'utils'
+import { getRpcProvider } from 'utils/rpc-provider'
+
+import { QuoteProvider, QuoteToken } from '../../interfaces'
+
+export interface FlashMintNavQuoteRequest {
+  chainId: number
+  isMinting: boolean
+  inputToken: QuoteToken
+  outputToken: QuoteToken
+  // In contrast to other quote providers, we always need the input amount (not the index amount)
+  inputTokenAmount: BigNumber
+  slippage: number
+}
+
+export interface FlashMintNavQuoteQuote {
+  inputTokenAmount: BigNumber
+  outputTokenAmount: BigNumber
+  reserveAssetSwapData: SwapDataV3
+}
+
+export class FlashMintNavQuoteProvider
+  implements QuoteProvider<FlashMintNavQuoteRequest, FlashMintNavQuoteQuote>
+{
+  constructor(
+    private readonly rpcUrl: string,
+    private readonly swapQuoteProvider: SwapQuoteProvider
+  ) {}
+
+  async getQuote(
+    request: FlashMintNavQuoteRequest
+  ): Promise<FlashMintNavQuoteQuote | null> {
+    const {
+      chainId,
+      inputToken,
+      inputTokenAmount,
+      isMinting,
+      outputToken,
+      slippage,
+    } = request
+
+    const indexToken = isMinting ? outputToken : inputToken
+    const usdc = USDC.address!
+
+    const swapQuoteRequest = {
+      chainId,
+      inputToken: isMinting ? inputToken.address : usdc,
+      outputToken: isMinting ? usdc : outputToken.address,
+      // TODO:
+      inputAmount: inputTokenAmount.toString(),
```

Comment:
@edkim for the reserve asset quote (to get the swap data) I guess it makes sense to use the `inputTokenAmount` of the quote request here when minting but what about redeeming? Because then `inputTokenAmount` would be the index token amount to redeem?

Reply:
When redeeming you need to swap the reserve asset (USDC) for the user's chosen output token (ex. WETH). So inputTokenAmount should be how much USDC we want to swap. Then 0x gives back a quote for how much WETH you'd get. 

To get the USDC value of however many setTokens the user is redeeming you can call this read function on the NAV Issuance Module: https://etherscan.io/address/0x2344674B23aD076908FD2396373CfE9cd48A1ba3#readContract#F2